java/Features.java

```diff
@@ -651,10 +652,12 @@ private static String getUnileverFeaturesHeader(){
             double avgArmAngel = AccStats.mean(fiveSecAvg);
 
             // 4. Absolute difference between successive values
-            double[] absoluteAvgDiff = computeAbsoluteDiff(fiveSecAvg);
-
-            // get the avg of difference
-            double avgArmAngelAbsDiff = AccStats.mean(absoluteAvgDiff);
+            //    ill-defined when we have fewer 10 sec for the epoch length
+            double avgArmAngelAbsDiff = 0;
+            if (x.length/sampleRate >= 10) {
+                double[] absoluteAvgDiff = computeAbsoluteDiff(fiveSecAvg);
+                avgArmAngelAbsDiff = AccStats.mean(absoluteAvgDiff);
+            }
```

Comment:
Shall we also explain this at the top of the function? Currently it says

`This won't work if the epoch length is less than 5-sec. So return 0 in that case.`

If I understand correctly `avgArmAngel` requires >=5, but `avgArmAngelDiff` requires >=10 right?

Reply:
Yes that's correct. Okay let me perhaps explain it a bit more.